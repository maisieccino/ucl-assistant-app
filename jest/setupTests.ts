import '@testing-library/jest-native/extend-expect'
import '@testing-library/react-native/cleanup-after-each'
import 'react-native-gesture-handler/jestSetup'

// Silence the warning: Animated: `useNativeDriver` is not supported because
// the native animated module is missing
// https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock(`react-native/Libraries/Animated/src/NativeAnimatedHelper`)
