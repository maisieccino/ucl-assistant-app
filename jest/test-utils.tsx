import { NavigationContainer } from '@react-navigation/native'
import { render as rtlRender, RenderAPI, act } from '@testing-library/react-native'
import React from 'react'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'
import thunk from "redux-thunk"
import debounce from "../lib/debounce"
import { initialState as defaultInitialState } from '../redux/reducers'

const middlewares = [
  debounce.middleware,
  thunk,
]
const mockStore = configureStore(middlewares)

const render = async (
  ui: React.ReactElement,
  {
    initialState = defaultInitialState,
    store = mockStore(initialState) as Store,
    ...renderOptions
  } = {},
): Promise<RenderAPI> => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <NavigationContainer>
        {children}
      </NavigationContainer>
    </Provider>
  )
  const result = rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  await act(async () => {})
  return result
}

export const waitForEventLoop = async (): Promise<void> => new Promise((resolve) => {
  const done = () => {
    jest.useFakeTimers()
    resolve()
  }
  jest.useRealTimers()
  setTimeout(done, 0)
})

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }
