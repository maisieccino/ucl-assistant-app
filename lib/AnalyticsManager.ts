import * as Amplitude from 'expo-analytics-amplitude'
import Constants from 'expo-constants'

import configureStore from "../configureStore"

const { store } = configureStore

const AMPLITUDE_API_KEY = Constants.manifest.extra
  && Constants.manifest.extra.amplitude
  && Constants.manifest.extra.amplitude.apiKey
  ? Constants.manifest.extra.amplitude.apiKey
  : null

const initialise = (): Promise<void> => Amplitude.initialize(AMPLITUDE_API_KEY)

const shouldTrackAnalytics = (): boolean => (
  AMPLITUDE_API_KEY
  && !__DEV__
  && store.getState().user.settings.shouldTrackAnalytics
)

const setUserId = (userId): void => {
  if (!shouldTrackAnalytics()) { return }

  // make sure this is unique
  if (userId) {
    Amplitude.setUserId(userId)
  }
}

const setUserProperties = ({ ...userProperties }): void => {
  if (!shouldTrackAnalytics()) { return }

  Amplitude.setUserProperties({ ...userProperties })
}

const clearUserProperties = (): void => {
  Amplitude.clearUserProperties()
}

const logEvent = (eventName: string, eventProperties?): void => {
  if (!shouldTrackAnalytics()) { return }

  if (eventProperties) {
    Amplitude.logEventWithProperties(eventName, eventProperties)
  } else {
    Amplitude.logEvent(eventName)
  }
}

const logScreenView = (screenName: string): void => {
  if (!shouldTrackAnalytics()) { return }

  logEvent(`VIEW_SCREEN_${screenName}`)
}

const events = {
  NOTIFICATIONS_ENABLE: `NOTIFICATIONS_ENABLE`,
  NOTIFICATIONS_SKIP: `NOTIFICATIONS_SKIP`,
  PUSH_NOTIFICATIONS_REGISTER: `PUSH_NOTIFICATIONS_REGISTER`,
  SETTINGS_GIVE_FEEDBACK: `SETTINGS_GIVE_FEEDBACK`,
  SETTINGS_RATE_APP: `SETTINGS_RATE_APP`,
  SETTINGS_VIEW_FAQS: `SETTINGS_VIEW_FAQS`,
}

export const AnalyticsManager = {
  clearUserProperties,
  events,
  initialise,
  logEvent,
  logScreenView,
  setUserId,
  setUserProperties,
}

export default AnalyticsManager
