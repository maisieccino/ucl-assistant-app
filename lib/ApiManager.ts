import axios from "axios"
import { encode as btoa } from "base-64"

import {
  EQUIPMENT_URL,
  FREE_ROOMS_URL,
  ROOMBOOKINGS_URL,
  ROOMS_URL,
  SITES_URL,
  TIMETABLE_URL,
  WORKSPACES_URL,
} from "../constants/API"
import ErrorManager from "./ErrorManager"
import LocalisationManager from './LocalisationManager'

const roomsController = {
  getBookings: async (
    token = null,
    {
      roomid,
      siteid,
      date = LocalisationManager.getMoment().format(`YYYYMMDD`),
    },
  ) => {
    if (!roomid || !siteid) {
      throw new Error(`Must specify roomid and siteid`)
    }
    try {
      const results = await axios.get(ROOMBOOKINGS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          date,
          roomid,
          siteid,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.bookings
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getEmptyRooms: async (
    token = null,
    {
      startDateTime = (new Date()).toISOString(),
      endDateTime = (
        LocalisationManager
          .getMoment()
          .add(1, `hours`)
          .toISOString()
      ),
    } = {},
  ) => {
    try {
      const results = await axios.get(FREE_ROOMS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          endDateTime,
          startDateTime,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.free_rooms
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getEquipment: async (token = null, { roomid, siteid }) => {
    if (!roomid || !siteid) {
      throw new Error(`Must specify roomid and siteid`)
    }
    try {
      const results = await axios.get(EQUIPMENT_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          roomid,
          siteid,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.equipment
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getSites: async (token = null) => {
    try {
      const results = await axios.get(SITES_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.sites
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  search: async (token = null, query) => {
    if (query && query.length < 3) {
      return {}
    }
    try {
      const results = await axios.get(ROOMS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          query,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.rooms
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
}

const workspacesApi = axios.create({
  baseURL: WORKSPACES_URL,
})

const workspacesController = {
  getLiveImage: async (token = null, { surveyId, mapId }) => {
    try {
      const result = await workspacesApi.get(`/getliveimage/map.svg`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          map_id: mapId,
          survey_id: surveyId,
        },
        responseType: `arraybuffer`,
      })
      const image = btoa(
        new Uint8Array(result.data).reduce(
          (data, byte) => data + String.fromCharCode(byte),
          ``,
        ),
      )
      // there may be no need to base64 SVGs
      // but I have yet to test this extensively
      // https://css-tricks.com/probably-dont-base64-svg/
      return `data:image/svg+xml;base64,${image}`
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getSummary: async (token = null) => {
    try {
      const { data, headers } = await workspacesApi.get(`/summary`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!data.content || data.error !== ``) {
        throw new Error(data.error || `There was a problem`)
      }
      return {
        data: data.content,
        lastModified: LocalisationManager.parseToMoment(
          headers[`last-modified`],
        ),
      }
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getWorkspaces: async (token = null) => {
    try {
      const { data, headers } = await workspacesApi.get(`/`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!data.content || data.error !== ``) {
        throw new Error(data.error)
      }
      return {
        data: data.content,
        lastModified: LocalisationManager.parseToMoment(
          headers[`last-modified`],
        ),
      }
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
}

const timetableApi = axios.create({
  baseURL: TIMETABLE_URL,
})

const timetableController = {
  getPersonalTimetable: async (
    token = null,
    date = LocalisationManager.getMoment(),
  ) => {
    const datePart = date
      ? `?date=${date.clone().startOf(`isoWeek`).format(`YYYY-MM-DD`)}`
      : ``
    const headers = {
      authorization: `Bearer ${token}`,
    }
    ErrorManager.addDetail({ headers })
    const results = await timetableApi.get(datePart, { headers })
    if (!results.data.content.ok || results.data.error !== ``) {
      throw new Error(
        results.data.error
        || `There was a problem fetching your personal timetable`,
      )
    }
    const { data: { content: { timetable } } } = results
    Object.keys(timetable).forEach((day) => {
      const dayTimetable = timetable[day]
      timetable[day] = {
        lastModified: LocalisationManager.parseToMoment(
          results.headers[`last-modified`],
        ),
        timetable: dayTimetable,
      }
    })
    return timetable
  },
  getPersonalWeekTimetable: async (
    token = null,
    date = LocalisationManager.getMoment().startOf(`isoWeek`),
  ) => {
    const queryParam = `?date=${date.format(`YYYY-MM-DD`)}`
    const headers = {
      authorization: `Bearer ${token}`,
    }
    ErrorManager.addDetail({ headers })
    const results = await timetableApi.get(`/week/${queryParam}`, {
      headers,
    })
    if (results.data.error !== ``) {
      throw new Error(
        results.data.error
        || `An error occurred when fetching your personal weekly timetable`,
      )
    }
    const { data: { content: { timetable } } } = results
    // eslint-disable-next-line sonarjs/no-identical-functions
    Object.keys(timetable).forEach((day) => {
      const dayTimetable = timetable[day]
      timetable[day] = {
        lastModified: LocalisationManager.parseToMoment(
          results.headers[`last-modified`],
        ),
        timetable: dayTimetable,
      }
    })
    return timetable
  },
}

export default {
  rooms: roomsController,
  timetable: timetableController,
  workspaces: workspacesController,
}
