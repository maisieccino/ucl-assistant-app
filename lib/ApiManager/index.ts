import PeopleController from './PeopleController'
import RoomsController from './RoomsController'
import TimetableController from './TimetableController'
import WorkspacesController from './WorkspacesController'

export default {
  people: PeopleController,
  rooms: RoomsController,
  timetable: TimetableController,
  workspaces: WorkspacesController,
}
