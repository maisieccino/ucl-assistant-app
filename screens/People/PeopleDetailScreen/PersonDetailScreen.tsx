import { RouteProp } from '@react-navigation/native'
import React, { useCallback } from "react"
import {
  ActivityIndicator, StyleSheet, View, ViewStyle,
} from "react-native"
import { connect, ConnectedProps } from "react-redux"
import { PeopleNavigatorParamList } from ".."
import Button from "../../../components/Button"
import {
  PaddedIcon,
  PageNoScroll,
  Spacer,
} from "../../../components/Containers"
import {
  BodyText,
  ButtonText,
  ErrorText,
  TitleText,
} from "../../../components/Typography"
import type { AppStateType } from "../../../configureStore"
import Colors from "../../../constants/Colors"
import { usePerson } from "../../../hooks"
import MailManager from "../../../lib/MailManager"

interface Style {
  container: ViewStyle,
  loadingContainer: ViewStyle,
}

const styles = StyleSheet.create<Style>({
  container: {
    flex: 1,
    paddingBottom: 40,
    paddingTop: 40,
  },
  loadingContainer: {
    alignItems: `center`,
    flex: 1,
    justifyContent: `center`,
  },
})

interface Props extends PropsFromRedux {
  // eslint-disable-next-line quotes
  route: RouteProp<PeopleNavigatorParamList, 'PeopleDetail'>,
}

export const PersonDetailScreen: React.FC<Props> = ({
  token,
  route: {
    params: {
      email,
    },
  },
}) => {
  const {
    status: fetchStatus,
    data: {
      name,
      status,
      department,
    } = {},
    error,
  } = usePerson(token, email)

  const sendEmail = useCallback(() => MailManager.composeAsync({ recipients: [email] }), [email])

  return (fetchStatus === `loading`) ? (
    <PageNoScroll style={styles.loadingContainer}>
      <ActivityIndicator size="large" />
    </PageNoScroll>
  ) : (fetchStatus === `error`) ? (
    <PageNoScroll style={styles.container}>
      {error && (
        <ErrorText>
          {`Error: ${error.name} ${error.message}`}
        </ErrorText>
      )}
    </PageNoScroll>
  ) : (
    <PageNoScroll>
      <View style={styles.container}>
        <TitleText>{name}</TitleText>

        <BodyText>
          {`${status}, ${department}`}
        </BodyText>
        <BodyText>
          {`Email: ${email}`}
        </BodyText>
        <Spacer />
        <Button onPress={sendEmail}>
          <PaddedIcon name="mail" size={24} color={Colors.pageBackground} />
          <ButtonText>Send Email</ButtonText>
        </Button>
      </View>
    </PageNoScroll>
  )
}

const connector = connect(
  (state: AppStateType) => ({
    token: state.user.token,
  }),
  () => ({}),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(PersonDetailScreen)
