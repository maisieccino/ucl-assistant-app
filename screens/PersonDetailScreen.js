import PropTypes from "prop-types"
import React, { Component } from "react"
import { ActivityIndicator } from "react-native"
import { connect } from "react-redux"

import { fetchPerson as fetchPersonAction } from "../actions/peopleActions"
import Button from "../components/Button"
import { PaddedIcon, PageNoScroll, Spacer } from "../components/Containers"
import {
  BodyText,
  ButtonText,
  ErrorText,
  TitleText,
} from "../components/Typography"
import Colors from "../constants/Colors"
import MailManager from "../lib/MailManager"

class PersonDetailScreen extends Component {
  static navigationOptions = {
    title: `Person Detail`,
  }

  static propTypes = {
    department: PropTypes.string,
    email: PropTypes.string,
    error: PropTypes.string,
    fetchPerson: PropTypes.func,
    isFetching: PropTypes.bool,
    /* eslint-disable react/no-unused-prop-types */
    name: PropTypes.string,
    navigation: PropTypes.shape().isRequired,
    route: PropTypes.shape().isRequired,
    status: PropTypes.string,
    token: PropTypes.string,
    /* eslint-enable react/no-unused-prop-types */
  }

  static defaultProps = {
    department: ``,
    email: ``,
    error: ``,
    fetchPerson: () => { },
    isFetching: false,
    name: ``,
    status: ``,
    token: ``,
  }

  static mapStateToProps = (state) => ({
    department: state.people.person.department,
    email: state.people.person.email,
    error: state.people.fetchError,
    isFetching: state.people.isFetching,
    name: state.people.person.name,
    status: state.people.person.status,
    token: state.user.token,
  })

  static mapDispatchToProps = (dispatch) => ({
    fetchPerson: (token, email) => dispatch(fetchPersonAction(token, email)),
  })

  constructor(props) {
    super(props)
    const { params } = props.route
    this.state = { ...params }
  }

  componentDidMount() {
    const { fetchPerson, token } = this.props
    const { email } = this.state
    fetchPerson(token, email)
  }

  sendEmail = () => {
    const { email } = this.state
    MailManager.composeAsync({
      recipients: [email],
    })
  }

  render() {
    const {
      name, status, department, email, isFetching, error,
    } = this.props
    return isFetching ? (
      <PageNoScroll>
        <ActivityIndicator size="large" />
      </PageNoScroll>
    ) : (
        <PageNoScroll>
          <TitleText>{name}</TitleText>
          {error.length > 0 && (
            <ErrorText>
              {`Error: ${error}`}
            </ErrorText>
          )}
          <BodyText>
            {`${status}, ${department}`}
          </BodyText>
          <BodyText>
            {`Email: ${email}`}
          </BodyText>
          <Spacer />
          <Button onPress={this.sendEmail}>
            <PaddedIcon name="mail" size={24} color={Colors.pageBackground} />
            <ButtonText>Send Email</ButtonText>
          </Button>
        </PageNoScroll>
    )
  }
}

export default connect(
  PersonDetailScreen.mapStateToProps,
  PersonDetailScreen.mapDispatchToProps,
)(PersonDetailScreen)
