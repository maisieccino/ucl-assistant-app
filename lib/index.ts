import AnalyticsManager from './AnalyticsManager'
import ApiManager from './ApiManager'
import AssetManager from './AssetManager'
import debounce from './debounce'
import DeviceManager from './DeviceManager'
import ErrorManager from './ErrorManager'
import LocalisationManager from './LocalisationManager'
import MailManager from './MailManager'
import MapsManager from './MapsManager'
import PushNotificationsManager from './PushNotificationsManager'
import Shadow from './Shadow'
import WebBrowserManager from './WebBrowserManager'

const Random = {
  array: (arr) => arr[Math.floor(Math.random() * arr.length)],
}

export {
  AnalyticsManager,
  ApiManager,
  AssetManager,
  DeviceManager,
  ErrorManager,
  LocalisationManager,
  MailManager,
  MapsManager,
  PushNotificationsManager,
  Random,
  Shadow,
  WebBrowserManager,
  debounce,
}
