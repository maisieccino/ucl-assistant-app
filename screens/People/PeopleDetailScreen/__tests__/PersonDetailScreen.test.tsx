import { RouteProp } from '@react-navigation/native'
import React from 'react'
import PersonDetailScreen from '..'
import { PeopleNavigatorParamList } from "../.."
import { fireEvent, render, wait } from '../../../../jest/test-utils'
import { ApiManager, MailManager } from "../../../../lib"
import type { Person } from '../../../../types/uclapi'

MailManager.composeAsync = jest.fn()

describe(`PersonDetailScreen`, () => {
  const mockPerson: Person = {
    department: `City of Town PD`,
    email: `j.cloth@city.town.uk`,
    name: `Jack Cloth`,
    status: `Detective Inspector`,
  }

  const props = {
    route: {
      params: {
        email: mockPerson.email,
      },
    // eslint-disable-next-line quotes
    } as RouteProp<PeopleNavigatorParamList, 'PeopleDetail'>,
    token: `supersecure`,
  }

  ApiManager.people.fetchPerson = jest.fn(() => Promise.resolve(mockPerson))

  const setup = () => render(<PersonDetailScreen {...props} />)

  beforeAll(() => {
    jest.useFakeTimers()
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it(`renders fine`, async () => {
    const wrapper = setup()
    const { getByText } = wrapper
    expect(ApiManager.people.fetchPerson).toBeCalledTimes(1)
    await wait()

    expect(getByText(new RegExp(mockPerson.department, `i`))).toBeTruthy()
    expect(getByText(new RegExp(mockPerson.name, `i`))).toBeTruthy()
    expect(getByText(new RegExp(mockPerson.status, `i`))).toBeTruthy()
    expect(getByText(new RegExp(mockPerson.email, `i`))).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it(`sends email`, async () => {
    const { getByText } = setup()
    const button = getByText(/send email/i)
    fireEvent.press(button)
    await wait()

    expect(MailManager.composeAsync).toBeCalledTimes(1)
    expect(MailManager.composeAsync).toBeCalledWith({ recipients: [mockPerson.email] })
  })
})
