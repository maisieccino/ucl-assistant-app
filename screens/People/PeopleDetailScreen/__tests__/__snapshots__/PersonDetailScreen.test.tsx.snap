// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetailScreen renders fine 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fefefe",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#fefefe",
          "flex": 1,
          "flexDirection": "column",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "paddingLeft": 20,
            "paddingRight": 20,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "paddingBottom": 40,
            "paddingTop": 40,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu-bold",
                "fontSize": 35,
                "marginBottom": 10,
                "marginTop": 1,
              },
              Object {},
            ]
          }
        >
          Jack Cloth
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu",
                "fontSize": 16,
                "marginBottom": 1,
                "marginTop": 1,
              },
              undefined,
            ]
          }
        >
          Detective Inspector, City of Town PD
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu",
                "fontSize": 16,
                "marginBottom": 1,
                "marginTop": 1,
              },
              undefined,
            ]
          }
        >
          Email: j.cloth@city.town.uk
        </Text>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279998859,
                  4283151357,
                ],
                "endPoint": Array [
                  1,
                  0,
                ],
                "locations": undefined,
                "startPoint": Array [
                  0,
                  1,
                ],
              }
            }
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "elevation": 2,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginTop": 5,
                  "minHeight": 55,
                  "minWidth": 150,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "shadowColor": "#1B998B",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.75,
                  "shadowRadius": 3,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fefefe",
                    "fontSize": 24,
                  },
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  Object {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï†Œ
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#fefefe",
                    "fontFamily": "apercu",
                    "fontSize": 20,
                    "marginBottom": 1,
                    "marginTop": 1,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Send Email
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
